ckanext/datastore/backend/postgres.py

```diff
@@ -934,6 +934,29 @@ def create_table(context, data_dict, plugin_data):
     sql_fields = u", ".join([u'{0} {1}'.format(
         identifier(f['id']), f['type']) for f in fields])
 
+    # (canada fork only): foreign keys
+    # TODO: upstream contrib!!
+    foreign_keys = data_dict.get('foreign_keys')
+    foreign_fields = []
+    if foreign_keys:
+        for f_table_name, column_name_map in foreign_keys.items():
+            fk_counter = 1
+            for p_column_name, f_column_name in column_name_map.items():
+                log.debug('Trying to build foreign key constraint for {0}.{1} on column {2}'.format(
+                    f_table_name,
+                    f_table_name,
+                    p_column_name
+                ))
+                foreign_fields.append('CONSTRAINT {0} FOREIGN KEY ({1}) REFERENCES {2}({3})'.format(
+                    identifier('fk_%s_%s' % (f_table_name, fk_counter)),
+                    identifier(p_column_name),
+                    identifier(f_table_name),
+                    identifier(f_column_name)
+                ))
+                fk_counter += 1
```

Comment:
something like?
```suggestion
            foreign_fields.append('CONSTRAINT {0} FOREIGN KEY ({1}) REFERENCES {2}({3})'.format(
                identifier('fk_%s_%s' % (f_table_name, fk_counter)),
                ','.join(identifier(k) for k in column_name_map.keys()),
                identifier(f_table_name),
                ','.join(identifier(v) for v in column_name_map.values())
            ))
            fk_counter += 1
```

Reply:
Thanks that worked great. Am able to make a foreign key constraint as long as the reference table has a primary key of course. and in PSQL it will allow to foreign key on unique key indices as well. E.g. the `primary_key` in datastore_create is only used in the unique index key. So it will still work on that.